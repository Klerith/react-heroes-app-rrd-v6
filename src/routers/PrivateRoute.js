import { useContext } from 'react';
import { Navigate, useLocation } from 'react-router-dom';
import { AuthContext } from '../auth/authContext';


export const PrivateRoute = ({ children }) => {

    const { user } = useContext(AuthContext);
    const { pathname, search } = useLocation();
    
    localStorage.setItem('lastPath', pathname + search );
    
    return user.logged
        ? children
        : <Navigate to="/login" />
}
