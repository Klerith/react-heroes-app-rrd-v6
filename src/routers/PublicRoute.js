import { useContext } from 'react';
import { Navigate } from 'react-router-dom';
import { AuthContext } from '../auth/authContext';



export const PublicRoute = ({ children }) => {

    const { user } = useContext(AuthContext);

    
    return user.logged
        ? <Navigate to="/marvel" />
        : children
}
