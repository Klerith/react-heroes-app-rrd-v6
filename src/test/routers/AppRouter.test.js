import { AppRouter } from '../../routers/AppRouter';
import { mount } from 'enzyme'
import { AuthContext } from '../../auth/authContext';


describe('Pruebas en <AppRouter />', () => {

    

    test('debe de mostrar el login si no está autenticado', () => {

        const contextValue = {
            user: {
                logged: false
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h1').text().trim() ).toBe( 'Login' );

        
    })
    
    test('debe de mostrar el componente de Marvel si está autenticado', () => {

        const contextValue = {
            user: {
                logged: true,
                name: 'Pepe'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        // console.log(wrapper.html());

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.navbar').exists() ).toBe( true );

        
    })
    
})


