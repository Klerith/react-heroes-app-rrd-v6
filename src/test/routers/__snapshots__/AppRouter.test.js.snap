// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el componente de Marvel si está autenticado 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        MarvelScreen
      </h1>
      <hr />
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Spider Man"
                  className="card-img"
                  src="/assets/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Captain America"
                  className="card-img"
                  src="/assets/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Iron Man"
                  className="card-img"
                  src="/assets/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Thor"
                  className="card-img"
                  src="/assets/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hulk"
                  className="card-img"
                  src="/assets/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Wolverine"
                  className="card-img"
                  src="/assets/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Daredevil"
                  className="card-img"
                  src="/assets/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hawkeye"
                  className="card-img"
                  src="/assets/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Cyclops"
                  className="card-img"
                  src="/assets/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img"
                  src="/assets/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Más...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no está autenticado 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
